### PR TITLE
Use X-Forwarded-Host header value if presign.host config is empty

### PR DESCRIPTION
**Pull request by tranchitella**
Changelog: none

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

**Comment by tranchitella** [comment]
Spurious failure: `tests/test_demo_artifact.py::TestDemoArtifact::test_demo_artifact`

File: api/http/api_deployments.go [hunk 1]
```diff
@@ -379,8 +378,16 @@ func (d *DeploymentsApiHandlers) DownloadConfiguration(w rest.ResponseWriter, r
...
 	hdr.Set("Content-Disposition", `attachment; filename="artifact.mender"`)
 	hdr.Set("Content-Type", app.ArtifactContentType)
+	artifactPayload, err := ioutil.ReadAll(artifact)
+	if err != nil {
+		// There's not anything we can do here in terms of the response.
+		rw.WriteHeader(http.StatusOK)
+		l.Error(err.Error())
+		return
+	}
+	hdr.Set("Content-Length", strconv.Itoa(len(artifactPayload)))
```
Comment (alfrunes) [inline api/http/api_deployments.go:388]: If you buffer the artifact in memory, it's not too late to report an internal error while reading it.
```suggestion

	artifactPayload, err := ioutil.ReadAll(artifact)
	if err != nil {
		d.view.RenderInternalError(e, r, err, l)
		return
	}
	hdr := rw.Header()
	hdr.Set("Content-Disposition", `attachment; filename="artifact.mender"`)
	hdr.Set("Content-Type", app.ArtifactContentType)
	hdr.Set("Content-Length", strconv.Itoa(len(artifactPayload)))
```

Comment (alfrunes) [reply api/http/api_deployments.go:388]: On the other hand, I believe `net/http` adds the content length header if the response is written in a single write.

**Review by alfrunes — CHANGES_REQUESTED** [review]

File: api/http/api_deployments.go [hunk 2]
```diff
@@ -379,8 +379,16 @@ func (d *DeploymentsApiHandlers) DownloadConfiguration(w rest.ResponseWriter, r
 	hdr := rw.Header()
 	hdr.Set("Content-Disposition", `attachment; filename="artifact.mender"`)
 	hdr.Set("Content-Type", app.ArtifactContentType)
+	artifactPayload, err := ioutil.ReadAll(artifact)
+	if err != nil {
+		// There's not anything we can do here in terms of the response.
+		rw.WriteHeader(http.StatusOK)
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply api/http/api_deployments.go:384]: No idea, this is what  the code was doing before...

[the remark this answers is not in the document]
Comment (tranchitella) [reply api/http/api_deployments.go]: You are right, makes no sense here. Rewritten as Internal Server Error.

File: api/http/api_deployments.go (hunk 1 above)
Comment (tranchitella) [reply api/http/api_deployments.go]: Right, reworked. Regarding the content-length, it is not there if we don't set it explicitly...

**Comment by tranchitella** [comment]
@alfrunes can you please have another look to this PR? We need to unblock the mender-configure client acceptance tests.

**Review by alfrunes — APPROVED** [review]
LGTM

**Pull request merged**